engine/common/provider/engine.go

```diff
@@ -3,22 +3,31 @@ package provider
 import (
 	"errors"
 	"fmt"
+	"math/rand"
 
 	"github.com/rs/zerolog"
 	"github.com/vmihailenco/msgpack"
 
 	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/engine/common/provider/internal"
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/model/flow/filter"
 	"github.com/onflow/flow-go/model/messages"
 	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/module/mempool"
 	"github.com/onflow/flow-go/module/metrics"
 	"github.com/onflow/flow-go/network"
 	"github.com/onflow/flow-go/network/channels"
 	"github.com/onflow/flow-go/state/protocol"
 	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/logging"
 )
 
+// DefaultRequestProviderWorkers is the default number of workers used to process entity requests.
+const DefaultRequestProviderWorkers = uint(100)
```

Comment:
100 seems like a lot. any idea what volume of requests we see today?

Reply:
Looking at `mainnet` data, we expect 3-5 requests on average. I think 5 is a safe number of workers hence: https://github.com/onflow/flow-go/pull/3404/commits/323fa27ab4a9d380ddc23b12b9ddfe25fc031316